src/theory/strings/theory_strings.cpp

```diff
@@ -3497,7 +3544,6 @@ TheoryStrings::ProcessLoopResult TheoryStrings::processLoop(
                      nm->mkNode(kind::REGEXP_STAR,
                                 nm->mkNode(kind::STRING_TO_REGEXP, restr))));
       cc = cc == d_true ? conc2 : nm->mkNode(kind::AND, cc, conc2);
-      d_regexp_ant[conc2] = ant;
```

Comment:
Why is it ok to remove this?

Reply:
This is a very old optimization. It appears that this map is storing an alternative way of explaining a regular expression. For example, `x ++ "ab" = "ba" ++ x` explains `x in ("ba")* ++ "b"`.

When untangling the regular expression solver, I found this map would need to be accessed by both the core strings solver and the regular expression solver. 

Since this optimization is not very important (and its correctness is questionable to me), I decided now is a good time to remove it.

Reply:
That makes sense, thanks for the explanation.